include/ringbuffer_utils.hpp

```diff
@@ -17,55 +17,78 @@ namespace ddprof {
 
 struct RingBufferInfo;
 
-class RingBufferWriter {
+// Return `x` rounded up to next multiple of `y`
+// `y` must be a power of 2
+// Return 0 for x==0 or x > std::numeric_limits<uint64_t>::max()-7
+constexpr inline uint64_t round_up_to_mutiple_of_pow2(uint64_t x, uint64_t y) {
```

Comment:
minor, naming is a little confusing, it is a multiple of `y`
Nice use of constexpr

Reply:
`constexpr` is not useful for my use here, but it does not hurt :)
Comments clearly explain what this function does. Is it the name of the function or the name of the `y` argument that bothers you ? Do you have an idea for a better name ?

Reply:
Indeed, the comment is good. 
I was just confused as to why the function would take two arguments with current naming, but I can't think of anything better. The pow2 is a pre-requisit of y, so the real naming would be something ugly like
`round_up_to_multiple_of_y_assuming_pow2`